addons/intel/e810.go

```diff
@@ -56,21 +68,87 @@ func OnPTPConfigChangeE810(nodeProfile *ptpv1.PtpProfile) error {
 	return nil
 }
 
-func PopulateHwConfigE810(hwconfigs *[]ptpv1.HwConfig) error {
+func AfterRunPTPCommandE810(data *interface{}, nodeProfile *ptpv1.PtpProfile, command string) error {
+	glog.Info("calling onPTPConfigChange for e810 plugin")
+	var e810Opts E810Opts
+	var err error
+	var optsByteArray []byte
+	var stdout []byte
+
+	e810Opts.EnableDefaultConfig = false
+
+	for name, opts := range (*nodeProfile).Plugins {
+		if name == "e810" {
+			optsByteArray, _ = json.Marshal(opts)
+			err = json.Unmarshal(optsByteArray, &e810Opts)
+			if err != nil {
+				glog.Error("e810 failed to unmarshal opts: " + err.Error())
+			}
+			for _, ublxOpt := range e810Opts.UblxConfigCmds {
+				ublxArgs := ublxOpt.E810UblxArgs
+				glog.Infof("Running cmd /usr/bin/ubxtool with args %s ", strings.Join(ublxArgs, ", "))
+				stdout, err = exec.Command("/usr/local/bin/ubxtool", ublxArgs...).CombinedOutput()
+				glog.Infof(string(stdout))
+			}
+			if command == "gpspipe" {
+				glog.Infof("AfterRunPTPCommandE810 doing ublx config for command: %s", command)
+				for _, ublxOpt := range e810Opts.UblxStatusCmds {
+					ublxArgs := ublxOpt.E810UblxArgs
+					glog.Infof("Running /usr/bin/ubxtool with args %s", strings.Join(ublxArgs, ", "))
+					stdout, err = exec.Command("/usr/local/bin/ubxtool", ublxArgs...).CombinedOutput()
+					//stdout, err = exec.Command("/usr/local/bin/ubxtool", "-p", "STATUS").CombinedOutput()
+					_data := *data
+					if data != nil {
+						glog.Infof("Saving status to hwconfig: %s", string(stdout))
+						var pluginData *E810PluginData = _data.(*E810PluginData)
+						_pluginData := *pluginData
+						statusString := fmt.Sprintf("ublx data: %s", string(stdout))
+						*_pluginData.hwplugins = append(*_pluginData.hwplugins, statusString)
+					} else {
+						glog.Infof("Not saving status to hwconfig: ", string(stdout))
+					}
+				}
+			} else {
+				glog.Infof("AfterRunPTPCommandE810 doing nothing for command: %s", command)
+			}
+		}
+	}
+	return nil
+}
+
+func PopulateHwConfigE810(data *interface{}, hwconfigs *[]ptpv1.HwConfig) error {
 	//hwConfig := ptpv1.HwConfig{}
 	//hwConfig.DeviceID = "e810"
 	//*hwconfigs = append(*hwconfigs, hwConfig)
+	if data != nil {
+		_data := *data
+		var pluginData *E810PluginData = _data.(*E810PluginData)
+		_pluginData := *pluginData
+		if _pluginData.hwplugins != nil {
+			for _, _hwconfig := range *_pluginData.hwplugins {
+				hwConfig := ptpv1.HwConfig{}
+				hwConfig.DeviceID = "e810"
+				hwConfig.Status = _hwconfig
+				*hwconfigs = append(*hwconfigs, hwConfig)
```

Comment:
if there is a hwconfig already exist with DeviceID=="e810", do we replace it or add another resulting in multiple hwconfigs with same DeviceID?